gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/FsBenchmark.java

```diff
@@ -0,0 +1,468 @@
+              + " --file=gs://${BUCKET}/path/to/test/object"
+              + " [--num-open=<number of file open>]"
+              + " [--read-size=<read buffer size in bytes>]"
+              + " [--num-reads=<number of random reads per file open>]"
+              + " [--num-threads=<number of threads to run test>]");
+      return 1;
+    }
+
+    Path testFile = new Path(args.get("--file"));
+
+    warmup(
+        args,
+        () ->
+            benchmarkRandomRead(
+                fs,
+                testFile,
+                /* numOpen= */ 5,
+                /* readSize= */ 1024,
+                /* numReads= */ 20,
+                /* numThreads= */ 5));
+
+    benchmarkRandomRead(
+        fs,
+        testFile,
+        parseInt(args.getOrDefault("--num-open", String.valueOf(1))),
+        parseInt(args.getOrDefault("--read-size", String.valueOf(1024))),
+        parseInt(args.getOrDefault("--num-reads", String.valueOf(100))),
+        parseInt(args.getOrDefault("--num-threads", String.valueOf(1))));
+
+    return 0;
+  }
+
+  private void benchmarkRandomRead(
+      FileSystem fs, Path testFile, int numOpen, int readSize, int numReads, int numThreads) {
+    System.out.printf(
+        "Running random read test that reads %d bytes from '%s' file %d times per %d open"
+            + " operations in %d threads%n",
+        readSize, testFile, numReads, numOpen, numThreads);
+
+    List<LongSummaryStatistics> openLatencyNsList = new ArrayList<>(numThreads);
+    List<LongSummaryStatistics> seekLatencyNsList = new ArrayList<>(numThreads);
+    List<LongSummaryStatistics> readLatencyNsList = new ArrayList<>(numThreads);
+    List<LongSummaryStatistics> closeLatencyNsList = new ArrayList<>(numThreads);
+
+    ExecutorService executor = Executors.newFixedThreadPool(numThreads);
+    CountDownLatch initLatch = new CountDownLatch(numThreads);
+    CountDownLatch startLatch = new CountDownLatch(1);
+    CountDownLatch stopLatch = new CountDownLatch(numThreads);
+    List<Future<?>> futures = new ArrayList<>(numThreads);
+    for (int i = 0; i < numThreads; i++) {
+      futures.add(
+          executor.submit(
+              () -> {
+                FileStatus fileStatus = fs.getFileStatus(testFile);
+                long fileSize = fileStatus.getLen();
+                long maxReadPositionExclusive = fileSize - readSize + 1;
+
+                LongSummaryStatistics openLatencyNs = newLongSummaryStatistics(openLatencyNsList);
+                LongSummaryStatistics seekLatencyNs = newLongSummaryStatistics(seekLatencyNsList);
+                LongSummaryStatistics readLatencyNs = newLongSummaryStatistics(readLatencyNsList);
+                LongSummaryStatistics closeLatencyNs = newLongSummaryStatistics(closeLatencyNsList);
+
+                ThreadLocalRandom random = ThreadLocalRandom.current();
+                byte[] readBuffer = new byte[readSize];
+
+                initLatch.countDown();
+                startLatch.await();
```

Comment:
Why not `awaitUninterruptibly` here but elsewhere?

Reply:
Because this it run inside the Callable that throws exception anyway, so `.await()` does not force you to add try/catch